taichi/struct/struct_opengl.h

```diff
@@ -31,6 +32,10 @@ class OpenglStructCompiler {
 
   std::vector<SNode *> snodes_;
   std::string src_code_;
+  std::map<std::string, size_t> stride_map_;
```

Comment:
If you don't need ordering of the keys, then prefer `std::unordered_map/std::unordered_set`, which are the actual hash tables in C++.

Reply:
Is there any application needs ordered `map`? Or `map` is just deprecated, should always use `unordered_map`?

Reply:
Use `map` when you need a red-black tree :)

Reply:
> See here: https://www.geeksforgeeks.org/map-vs-unordered_map-c/
> 
> Long story short: use `map` when you need a key order.